bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -1,4 +1,8 @@
 /**
```

Comment:
There are MANY changes in this class - is this really all related to this feature or does it include some further refactorings? In the latter case, I would prefer to have it split into separate PRs, which would make reviews much easier.